client/ayon_resolve/plugins/create/create_editorial_package.py

```diff
@@ -1,88 +1,120 @@
+                                           pre_create_data)
 
         current_timeline = lib.get_current_timeline()
 
-        context = ask_for_context(
-            project_name, current_folder["id"]
-        )
-
-        if context is None:
-            return
-
-        # Get workfile path to save to.
-        project_name = context["project_name"]
-        folder = get_folder_by_id(project_name, context["folder_id"])
-
-        # task is optional so we need to check if it is set
-        task = None
-        if "task_id" in context:
-            task = get_task_by_id(project_name, context["task_id"])
-
-        # reset self.data to be pointing in the set context data
-        self.data["folderPath"] = folder["path"]
-
-        if task:
-            self.data.update({
-                "taskId": task["id"],
-                "taskName": task["name"],
-            })
-
         if not current_timeline:
-            raise RuntimeError("Make sure to have an active current timeline.")
+            raise CreatorError("Make sure to have an active current timeline.")
 
         timeline_media_pool_item = lib.get_timeline_media_pool_item(
             current_timeline
         )
 
-        publish_data = deepcopy(self.data)
+        publish_data = deepcopy(instance_data)
+
         # add publish data for streamline publishing
         publish_data["publish"] = get_editorial_publish_data(
-            folder_path=folder["path"],
-            product_name=self.data["productName"],
+            folder_path=instance_data["folderPath"],
+            product_name=self.product_name,
         )
 
+        publish_data["label"] = current_timeline.GetName()
         timeline_media_pool_item.SetMetadata(
-            lib.pype_tag_name, json.dumps(publish_data)
+            constants.AYON_TAG_NAME, json.dumps(publish_data)
         )
 
+        publish_data["media_pool_item_id"] = timeline_media_pool_item.GetUniqueId()
+        new_instance = CreatedInstance(
+            self.product_type,
+            self.product_name,
+            publish_data,
+            self,
+        )
+        new_instance.transient_data["timeline_item"] = timeline_media_pool_item
+        self._add_instance_to_context(new_instance)
+
+    def collect_instances(self):
+        """Collect all created instances from current timeline."""
+        for media_pool_item in lib.iter_all_media_pool_clips():
+            data = media_pool_item.GetMetadata(constants.AYON_TAG_NAME)
+            if not data:
+                continue
+
+            try:
+                data = json.loads(data)
+            except json.JSONDecodeError:
+                self.log.warning(
+                    f"Failed to parse json data from media pool item: "
+                    f"{media_pool_item.GetName()}"
+                )
+                continue
+
+            # exclude all which are not productType editorial_pkg
+            if (
+                data.get("publish", {}).get("productType") != "editorial_pkg"
```

Comment:
Best to use the class variable here?
```suggestion
                data.get("publish", {}).get("productType") != self.product_type
```